scripts/analysis.py

```diff
@@ -250,229 +116,198 @@ def get_sim_time_duration(cursor):
 
 
-def get_from_facility(cursor, facility, duration, resources):
-    """ Returns timeseries list of quantity out of facility type
+def get_isotope_transactions(resources, compositions):
+    """Creates a dictionary with isotope name, mass, and time
 
     Parameters
     ----------
-    cursor: sqlite cursor
-        sqlite cursor
-    facility: str
-        name of facility type
-    duration: int
-        duration of simulation
-    resrources: list
-        list of lists, of quantity, senderid and time
+    resources: list
+        resource data from the resources table
+        list[0]: time
+        list[1]: sum(quantity)
+        list[2]: qualid
+    compositions: list
+        composition data from the compositions table
+        list[0]: qualid
+        list[1]: nucid
+        list[2]: massfrac
 
     Returns
     -------
-    list
-    timeseries list of mass outflux from facility type
+    transactions: dictionary
+        dictionary with keys as isotope and value as
+        list of tuples (mass moved, time)
     """
+    transactions = collections.defaultdict(list)
+    for res in resources:
+        for comp in compositions:
+            # res_qualid = res[2]
+            # comp_qualid = comp[0]
+            if res[2] == comp[0]:
+                # comp_nucid = comp[1]
+                # res_quantity = res[1s
+                # mass_frac = comp[2]
+                # res_time = res[0]
+                transactions[comp[1]].append((res[0], res[1] * comp[2]))
 
-    quantity = 0
-    timeseries = []
-    agentid = get_agent_id_from_prototype(cursor, facility)
-    for i in range(0, duration):
-        indices = [x for x, y in enumerate(resources) if y[2] == i]
-        for index in indices:
-            try:
-                if str(resources[index][1]) in agentid:
-                    quantity += resources[index][0]
-            except:
-                print('none in this timestep' + str(i))
-        timeseries.append(quantity/1000)
-
-    return timeseries
+    return transactions
 
 
-def total_waste_timeseries(cursor):
-    """Plots a stacked bar chart of the total waste mass vs time
+def commodity_in_out_facility(cursor, facility, commod_list,
+                              is_outflux, is_prototype, do_isotopic):
+    """ Returns timeseries of commodity in/outflux from facility or prototype
 
     Parameters
     ----------
     cursor: sqlite cursor
         sqlite cursor
+    facility: str
+        facility type as shown in spec or prototype name
+    commod_list: list
+        list of commodities
+    is_outflux: bool
+        gets outflux if True, influx if False
+    is_prototype: bool
+        searches using prototype name if True,
+        facility type if False
+    do_isotopic: bool
+        gets isotopic of outflux commodity if True
 
     Returns
     -------
-    null
-    stacked bar chart of waste mass vs time
+    commodity_dict: dictionary
+        dictionary with key: commodity, and value: timeseries list
```

Comment:
timeseries of what? I assume timeseries of masses of that commodity. If so, say what units. Also, Is the timeseries really a list? It must be quite full of zeros. It may be advisable to use a sparse dictionary rather than a list.

```
        dictionary with key=commodity, and value=timeseries of masses in kg
```